### PR TITLE
chore: setup Nx Cloud Id for PAT support

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Setup Nx Cloud Id for PAT support
See https://nx.dev/ci/recipes/security/personal-access-tokens for details

## Related issues

*- No issue associated -*

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -515,5 +515,14 @@
...
   },
   "useDaemonProcess": false,
-  "cacheDirectory": ".cache/nx"
+  "cacheDirectory": ".cache/nx",
+  "nxCloudId": "63e6951e043da20dd8321aec",
+  "tasksRunnerOptions": {
+    "default": {
+      "runner": "nx-cloud",
+      "options": {
+        "nxCloudId": "63e6951e043da20dd8321aec"
```
Comment (fpaul-1A) [inline nx.json:524]: according to the docs, we should add `nx-cloud` devDependency for it to work on Nx <= 19.6 (https://github.com/nrwl/nx/issues/28414 :'( )

Comment (kpanot) [reply nx.json:524]: Not sure to understand where in the doc it flags `nx-cloud` as mandatory devDep (and it is working locally today without it).
`nx-cloud` is convenient to register global on your machine but it is not mandatory in the dev in the repo (as the examples in the docs are `npx nx-cloud ...` based

Comment (fpaul-1A) [reply nx.json:524]: image

Comment (kpanot) [reply nx.json:524]: Indeed :S
It's working without it today but let's add it then

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -21,11 +21,14 @@ on:
...
 env:
   YARN_ENABLE_HARDENED_MODE: 0
+  NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
 
 jobs:
   # Check the integrity of yarn lock
   yarn_lock_check:
     runs-on: ubuntu-latest
+    env:
+      NX_SKIP_NX_CACHE: true
```
Comment (fpaul-1A) [inline .github/workflows/main.yml:31]: Not related to your change but don't we want the Nx cache for this step?

**Review by fpaul-1A — APPROVED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (kpanot) [reply .github/workflows/main.yml:31]: It is requires by `nx-cloud` runner when run in CI.
cf: https://github.com/AmadeusITGroup/otter/actions/runs/11702511447/job/32590842412

**Review by cpaulve-1A — APPROVED** [review]

**Pull request merged**